From 3606b66cc3d0bc9214f3875db3f69acdf0c2e65f Mon Sep 17 00:00:00 2001
From: Armin Novak <armin.novak@thincast.com>
Date: Wed, 11 Apr 2018 08:33:05 +0200
Subject: [PATCH 11/13] Fix rdpsnd channel detached handling.

Do not play samples if detached, but handle every other message
just like with an attached channel.
---
 channels/rdpsnd/client/rdpsnd_main.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/channels/rdpsnd/client/rdpsnd_main.c b/channels/rdpsnd/client/rdpsnd_main.c
index 0929360..9af8a72 100644
--- a/channels/rdpsnd/client/rdpsnd_main.c
+++ b/channels/rdpsnd/client/rdpsnd_main.c
@@ -497,7 +497,7 @@ static UINT rdpsnd_recv_wave_pdu(rdpsndPlugin* rdpsnd, wStream* s)
 	WLog_Print(rdpsnd->log, WLOG_DEBUG, "Wave: cBlockNo: %"PRIu8" wTimeStamp: %"PRIu16"",
 	           rdpsnd->cBlockNo, rdpsnd->wTimeStamp);
 
-	if (rdpsnd->device)
+	if (rdpsnd->device && rdpsnd->attached)
 	{
 		wStream* pcmData = StreamPool_Take(rdpsnd->pool, 4096);
 
@@ -588,11 +588,6 @@ static UINT rdpsnd_recv_pdu(rdpsndPlugin* rdpsnd, wStream* s)
 	Stream_Seek_UINT8(s); /* bPad */
 	Stream_Read_UINT16(s, BodySize);
 
-	if (!rdpsnd->attached)
-		goto out;
-
-	//WLog_ERR(TAG,  "msgType %"PRIu8" BodySize %"PRIu16"", msgType, BodySize);
-
 	switch (msgType)
 	{
 		case SNDC_FORMATS:
-- 
1.8.3.1

